terracotta/bootstrap/src/main/java/org/terracotta/quartz/DefaultClusteredJobStore.java

```diff
@@ -825,6 +825,33 @@ public void clearAllSchedulingData() throws JobPersistenceException {
 
   /**
    * <p>
+   * Reset the current state of the identified <code>{@link Trigger}</code>.
+   * </p>
+   *
+   * @see Trigger.TriggerState
+   */
+  @Override
+  public void resetTriggerState(org.quartz.TriggerKey key) throws JobPersistenceException {
+    lock();
+    try {
+      TriggerWrapper tw = triggerFacade.get(key);
+
+      // does the trigger exist?
+      if (tw == null) { return; }
+
+      tw.setState(TriggerState.WAITING, terracottaClientId, triggerFacade);
+
+      applyMisfire(tw);
```

Comment:
It is superfluous - will get detected and handled later if the misfire condition exists, but it is also likely that the trigger has actually misfired, so there is a small argument for eagerly updating it here (and the other locations).